src/mlx5_regex.c

```diff
@@ -468,6 +468,15 @@ mlx5_regex_register_read(struct ibv_context *ctx, int engine_id,
 	return 0;
 }
 */
+
+/* Check if Linux distribution supports the MAP_HUGE_SHIFT define. */
+/* We will allocate from the 2M hugepage pool.                     */
+#ifdef MAP_HUGE_SHIFT
+#define MLX_REGEX_MAP_HUGE_2MB (21 << MAP_HUGE_SHIFT)
```

Comment:
Should we not use the provided define for this? `#define MAP_HUGE_2MB    (21 << MAP_HUGE_SHIFT)` From: https://man7.org/linux/man-pages/man2/mmap.2.html

I think it's best to not duplicate effort thats already done for us. and then i would also reccommend simplifying the expression by masking the flag early like so:
```
    #ifdef MAP_HUGE_2MB
    #define MLX_REGEX_MAP_HUGETLB (MAP_HUGETLB | MAP_HUGE_2MB)
    #else 
    #define MLX_REGEX_MAP_HUGETLB (MAP_HUGETLB)
    #endif

    ...
    ctx->db_ctx[engine_id].mem_desc.ptr = mmap(NULL, db_size, PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANONYMOUS | MAP_POPULATE | MLX_REGEX_MAP_HUGETLB, -1, 0);
    
```

Reply:
Hi Robert, I'm not 100% sure if the MAP_HUGE_2MB define is available in all OSs - I will merge the commit and come back to this, as no time to retry on all OSs. I have a  vague memory that it wasn't available when investigating the huge pages issue initially, and the MAP_HUGE_SHIFT define was.